### PR TITLE
labels from strings to ints for in-memory build

### PR DESCRIPTION
**Pull request by varunsivashankar**
- [ ] Does this PR have a descriptive title that could go in our release notes?
- [ ] Does this PR add any new dependencies?
- [ ] Does this PR modify any existing APIs?
   - [ ] Is the change to the API backwards compatible?
- [ ] Should this result in any changes to our documentation, either updating existing docs or adding new ones?
 
#### What does this implement/fix? Briefly explain your changes.
For in-memory build, this PR changes how labels are processed internally from strings to ints. This increases the speed of checking label set intersections between two points, resulting in faster build time and higher QPS at search time.

#### Any other comments?
Code also written by Siddharth Gollapudi

### CODE REVIEW

File: include/index.h [hunk 1]
```diff
@@ -25,6 +25,8 @@
 #define DEFAULT_MAXC 750
 
 namespace diskann {
+    typedef unsigned label;
```
Comment (darvg) [inline include/index.h:28]: I recommend moving this typedef into `utils.h`: we may want to use it in places where `index.h` is not imported

File: include/index.h [hunk 2]
```diff
@@ -239,17 +241,17 @@ namespace diskann {
                                               float                *distances,
                                               InMemQueryScratch<T> *scratch,
         bool               use_filters = false,
-        const std::string &filter_label = std::string());
+        const label &filter_label = -1);
```
Comment (darvg) [inline include/index.h:244]: `label` is an unsigned: maybe use `UINT_MAX` instead of `-1`?

File: tests/search_memory_index.cpp [hunk 3]
```diff
@@ -56,8 +56,11 @@ int search_memory_index(diskann::Metric& metric, const std::string& index_path,
   }
 
   bool filtered_search = false;
-  if (filter_label != "")
+  diskann::label filter_label_as_num = -1;
```
Comment (darvg) [inline tests/search_memory_index.cpp:59]: same issue here: `diskann::label` is an unsigned: use `UINT_MAX` or some other appropriate value

**Comment by darvg** [comment]
also run a clang format, there seems to be a few indentation issues

File: src/index.cpp [hunk 4]
```diff
@@ -489,16 +489,17 @@ namespace diskann {
           std::istringstream iss(line);
           _u32               cnt = 0;
           _u32               medoid = 0;
-          std::string        label;
+          label        label;
```
Comment (darvg) [inline src/index.cpp:492]: A small indentation issue: run `clang-format` on all files included in the PR

**Review by darvg — APPROVED** [review]
lgtm

File: src/index.cpp [hunk 5]
```diff
@@ -511,7 +512,7 @@ namespace diskann {
       if (file_exists(universal_label_file)) {
         std::ifstream universal_label_reader(universal_label_file);
         universal_label_reader >> _universal_label;
-        _use_universal_label = true;
+        _use_universal_label = true; // to check
```
Comment (dkaczynski) [inline src/index.cpp]: What is `// to check` referring to here?

Comment (varunsivashankar) [reply src/index.cpp]: This was to try debugging something with the disk build. Not relevant to in-memory

Comment (dkaczynski) [reply src/index.cpp]: I'd recommend either being more explicit in the comment or removing it if it's no longer relevant

Comment (varunsivashankar) [reply src/index.cpp]: done

**Review by dkaczynski — APPROVED** [review]

**Pull request merged**